internal/graph/cycle_check_resolver.go

```diff
@@ -27,13 +29,20 @@ func (c *CycleDetectionCheckResolver) ResolveCheck(
 	ctx context.Context,
 	req *ResolveCheckRequest,
 ) (*ResolveCheckResponse, error) {
+	ctx, span := tracer.Start(ctx, "ResolveCheck")
+	defer span.End()
```

Comment:
I am starting to get worried about the number of spans that we are generating. It looks as though if the Check involves N subproblems we are generating, N * (number of resolvers) spans. If you have caching & singleflight turned on, that will be 4N. 

And this doesn't include database reads spans, which could be thousands.

Reply:
Yes we need a new span for this because otherwise it is inconsistent with the other resolvers. 

The concern is fair but I'm not sure why it is arising now; a single check has always had the potential to create a large number of spans. I'd prefer if we focused on just the changes in this PR. If you're able to quantify the cost of your concerns maybe we can address as a team soon.

Reply:
> Yes we need a new span for this because otherwise it is inconsistent with the other resolvers. 

Then i'd argue that we need to update the other resolvers too so that they do not create new spans. The number of spans has been a concern forever, specifically since we executed the [1 million RPS test](https://auth0.com/blog/getting-unlimited-scalability-with-okta-fine-grained-authorization/).

Reply:
If we want to holistically address the concern of too many spans per Check, then we should embark on a focused effort for that and come up with what we want our trace generation to look like under the umbrella of that effort.

If the objective of our traces is to visualize everything, then I think we're going to be hard pressed for reducing our span generation. However, if the purpose of our tracing is to gain insights into latent paths, then maybe should be more selective about where we capture trace information. In any case, this conversation seems better had outside of the scope of this PR, because it is already affecting us today.

Reply:
Right. The concern is fair but if we want observability into our check algorithm, we're going to need spans. I'm not convinced that the cost of this is not trivial and if it becomes a performance issue, we can address holistically within another body of work.